### PR TITLE
Remove normative language from proxy configuration and rework referen…

### LINKED ISSUE #451: Proxy configuration table uses normative language
In chapter [6. Capabilities](https://w3c.github.io/webdriver/webdriver-spec.html#capabilities#h-capabilities), section [6.1 Proxy](https://w3c.github.io/webdriver/webdriver-spec.html#capabilities#h-proxy)::
> The proxy configuration capability is a JSON Object nested within the primary capabilities. Implementations may define additional proxy configuration options, but they must not alter the semantics of those listed below. 

… and the following proxy configuration table uses normative language (MUST and SHOULD) which is not appropriate. The normative language should be saved for the algorithm that defines the steps to take to implement parsing it.

### PR DESCRIPTION
**Pull request by AutomatedTester**
…ces. Fixes #451

This is a general cleanup of the proxy configuration section of the spec.

---
This change is [Reviewable](https://reviewable.io/reviews/w3c/webdriver/563)

### CODE REVIEW

File: webdriver-spec.html [hunk 1]
```diff
@@ -450,6 +450,11 @@
    <!-- ScrollIntoViewOptions --> <li><dfn><a href=https://drafts.csswg.org/cssom-view/#dictdef-scrollintoviewoptions><code>ScrollIntoViewOptions</code></a></dfn>
   </ul>
 
+ <dt>Socks Proxy and related specification:
```
Comment (andreastt) [inline webdriver-spec.html]: s/Socks Proxy/SOCKS proxy/

File: webdriver-spec.html [hunk 2]
```diff
@@ -450,6 +450,11 @@
...
+ <dd><p>To be <dfn>Socks Proxy</dfn> and <dfn>Socks authentication</dfn> compliant,
```
Comment (andreastt) [inline webdriver-spec.html]: s/Socks/SOCKS/g

File: webdriver-spec.html [hunk 3]
```diff
@@ -1702,125 +1709,123 @@
...
 <tr>
-<td>"<code>proxyType</code>"
+<td><dfn>"<code>proxyType</code>"</dfn>
 <td>string
-<td>Indicates the type of proxy configuration. This value <em>SHOULD</em> be one
-of the following: "<code>pac</code>", "<code>noproxy</code>",
-"<code>autodetect</code>", "<code>system</code>", or "<code>manual</code>".
+<td>Indicates the type of proxy configuration.
+<td>"<code>pac</code>", "<code>noproxy</code>",
+"<code>autodetect</code>", "<code>system</code>", or "<code>manual</code>"
```
Comment (andreastt) [inline webdriver-spec.html]: Missing full stop.

**Comment by AutomatedTester** [comment]
landed in 9c8b5a6

**Pull request closed without merging**